programs/marginfi/src/instructions/marginfi_group/liquid_insurance_fund/deposit.rs

```diff
@@ -0,0 +1,160 @@
+    /// This is the insurance vault of the underlying bank
+    #[account(
+        mut,
+        seeds = [
+            INSURANCE_VAULT_SEED.as_bytes(),
+            bank.key().as_ref(),
+        ],
+        bump = bank.load()?.insurance_vault_bump
+    )]
+    pub bank_insurance_vault: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        seeds = [
+            LIQUID_INSURANCE_MINT_AUTHORITY_SEED.as_bytes(),
+            bank.key().as_ref(),
+        ],
+        bump
+    )]
+    pub mint_authority: AccountInfo<'info>,
+
+    #[account(
+        seeds = [
+            LIQUID_INSURANCE_MINT_SEED.as_bytes(),
+            bank.key().as_ref(),
+        ],
+        bump,
+    )]
+    pub mint: Account<'info, Mint>,
+
+    pub token_program: Program<'info, Token>,
+}
+
+pub fn deposit_into_liquid_insurance_fund(
+    ctx: Context<DepositIntoLiquidInsuranceFund>,
+    amount: u64,
+) -> MarginfiResult {
+    let DepositIntoLiquidInsuranceFund {
+        marginfi_group: marginfi_group_loader,
+        marginfi_account: marginfi_account_loader,
+        liquid_insurance_fund,
+        signer,
+        signer_token_account,
+        bank,
+        bank_insurance_vault,
+        mint_authority,
+        mint,
+        token_program,
+        ..
+    } = ctx.accounts;
+
+    check!(
+        !ctx.accounts
+            .marginfi_account
+            .load()?
+            .get_flag(DISABLED_FLAG),
+        MarginfiError::AccountDisabled
+    );
+
+    let mut liquid_insurance_fund = ctx.accounts.liquid_insurance_fund.load_mut()?;
+
+    // Get amount inside the bank's insurance vault
+    let total_bank_insurance_vault_amount = ctx.accounts.bank_insurance_vault.amount;
+
+    // update shares of the liquid insurance fund
+    liquid_insurance_fund.deposit_shares(
+        I80F48::from_num(amount),
+        I80F48::from_num(total_bank_insurance_vault_amount),
+    )?;
+
+    // Deposit user funds into the relevant insurance vault
+    liquid_insurance_fund.deposit_spl_transfer(
+        amount,
+        Transfer {
+            from: ctx.accounts.signer_token_account.to_account_info(),
+            to: ctx.accounts.bank_insurance_vault.to_account_info(),
+            authority: ctx.accounts.signer.to_account_info(),
+        },
+        token_program.to_account_info(),
+    )?;
+
+    // Send the user a tokenized representation of their deposit as shares
+    let user_deposited_share_amount = liquid_insurance_fund.get_shares(I80F48::from_num(amount))?;
+    let user_deposited_share_amount = user_deposited_share_amount
+        .checked_to_num::<u64>()
+        .ok_or(MarginfiError::MathError)?;
+
+    // mint tokens from mint and send to the senders token account
+    mint_to(
+        CpiContext::new_with_signer(
+            ctx.accounts.token_program.to_account_info(),
+            MintTo {
+                authority: ctx.accounts.mint_authority.to_account_info(),
+                to: ctx.accounts.signer_token_account.to_account_info(),
```

Comment:
how can this be the same token account? There must be two. One for the insured asset and one for the insurance shares. Also, do we really want to tokenize it? Has tax implications

Reply:
Yes -- tokenization will be removed for now for legal reasons